runhouse/resources/hardware/on_demand_cluster.py

```diff
@@ -464,6 +467,38 @@ def teardown(self):
         Example:
             >>> rh.ondemand_cluster("rh-cpu").teardown()
         """
+        # TODO [SB]: remove the den_auth check once we will get status of clusters without den_auth as well.
+        if self.den_auth:
+            try:
+                # cluster_config = self.config()
+                cluster_status_data = self.status()
+                status_data = {
+                    "status": "terminated",
+                    "resource_type": cluster_status_data.get("cluster_config").get(
+                        "resource_type"
+                    ),
+                    "data": cluster_status_data,
+                }
+                cluster_uri = rns_client.format_rns_address(self.rns_address)
+                api_server_url = cluster_status_data.get("cluster_config").get(
+                    "api_server_url", rns_client.api_server_url
+                )
+                post_status_data_resp = requests.post(
+                    f"{api_server_url}/resource/{cluster_uri}/cluster/status",
+                    data=json.dumps(status_data),
+                    headers=rns_client.request_headers(),
+                )
+                assert post_status_data_resp.status_code == 200
+            except Exception:
+                warning_msg = "Failed to update in Den that the cluster is terminated."
```

Comment:
any specific motivation for storing the warning text as a separate variable?

Reply:
Just for the sake of code reusability: when handling an error where status_code != 200 or when catching an exception, the prefix of the warning message is the same, but the suffix is different.

Don't have strong opinion on this though, can remove this variable.